src/Transport/Addressing/Sanitization/Strategies/EndpointOrientedTopologySanitization.cs

```diff
@@ -4,21 +4,13 @@
 
     public class EndpointOrientedTopologySanitization : ISanitizationStrategy
     {
-        IValidationStrategy validationStrategy;
-
-        public EndpointOrientedTopologySanitization(IValidationStrategy validationStrategy)
-        {
-            this.validationStrategy = validationStrategy;
-        }
-
-        public string Sanitize(string entityPathOrName, EntityType entityType)
+        public string Sanitize(string entityPathOrName, EntityType entityType, ValidationResult validationResult)
         {
             // remove invalid characters
             var rgx = new Regex(@"[^a-zA-Z0-9\-\._]");
             entityPathOrName = rgx.Replace(entityPathOrName, "");
 
-            var validationResult = validationStrategy.IsValid(entityPathOrName, entityType);
-            if (!validationResult.IsValid)
+            if (!validationResult.LengthIsValid)
```

Comment:
This logic seems different than in the current release: `validationResult` is the result produced by `ValidationStrategy` and is created when `AddressingLogic` invokes validation. Inside `SanitizationStrategy` we use provided value without executing again validation to evaluate if length is fine after removing invalid chars.
In the current release transport checks again validity after removing invalid chars

Is it true?